### PR TITLE
tasks 19, 20, final

### PR DESCRIPTION
**Pull request by Transcendentall**

### CODE REVIEW

File: bolychev.li/final.hs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    case list of
+        [] -> []
+        x:xs -> functionInverse xs ++ [x]
+
+
+functionOutput :: [Int] -> [Int]
+functionOutput list = list
+
+functionCount :: [Int] -> [(Int, Int)]
+functionCount list = map (\x -> (x, length (filter (==x) list))) list
```
Comment (DieNice) [inline bolychev.li/final.hs:20]: Prelude> functionCount list = map (\x -> (x, length (filter (==x) list))) list
Prelude> functionCount [1,1,2,3,3,3,4,4,4,4,0]
[(1,2),(1,2),(2,1),(3,3),(3,3),(3,3),(4,4),(4,4),(4,4),(4,4),(0,1)]

File: bolychev.li/final.hs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+functionSublist :: Eq a => [a] -> [a] -> Bool
+functionSublist a b = all (`elem` b) a
```
Comment (DieNice) [inline bolychev.li/final.hs:23]: Prelude> functionCount list = map (\x -> (x, length (filter (==x) list))) list
Prelude> functionCount [1,1,2,3,3,3,4,4,4,4,0]
[(1,2),(1,2),(2,1),(3,3),(3,3),(3,3),(4,4),(4,4),(4,4),(4,4),(0,1)]
Prelude> functionSublist a b = all (`elem` b) a
Prelude> functionSublist [1,2,3,4,5,6,7,0] [2,3,4]
False
Prelude> functionSublist [1,2,3,4,5,6,7,0] [2,3]
False
Prelude> functionSublist [1,2,3,4,5,6,7,0] [1,2,3]
False
Prelude> functionSublist [1,2,3,4,5,6,7,0] []
False
Prelude> functionSublist [1,2,3,4,5,6,7,0] [7,0]
False
Prelude> functionSublist [1,2,3,4,5,6,7,0] [0]
False
Prelude> functionSublist [1,2,3,4,5,6,7,0] [1,2,3,4,5,6,7,0]
True
Prelude> functionSublist [1,2,3,4,5,6,7,0] [2,3,4,5,6,7,0]
False
Prelude> functionSublist [1,2,3,4,5,6,7,0] [3,4,5,6,7,0]
False
Prelude> functionSublist [1,2,3,4,5,6,7,0] [4,5,6,7,0]
False
Prelude> functionSublist [1,2,3,4,5,6,7,0] [5,6,7,0]
False
Prelude> functionSublist [1,2,3,4,5,6,7,0] [6,7,0]
False
Prelude> functionSublist [1,2,3,4,5,6,7,0] [7,0]

File: bolychev.li/final.hs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+functionSublistBeginningElement :: Int -> [Int] -> [Int]
+functionSublistBeginningElement a list = (\(x,y) -> if ((a-1) <= y) then [x] else []) =<< zip list [0..]
```
Comment (DieNice) [inline bolychev.li/final.hs:26]: По элементу а не по индексу

File: bolychev.li/final.hs (hunk 1 above)
Comment (Transcendentall) [reply bolychev.li/final.hs:20]: functionCount :: Eq a => [a] -> [(a, Int)]
functionCount xs = map count ys
    where ys = nub xs
          count x = (x, foldl (count' x) 0 xs)
          count' x acc y = (if x == y then (1+) else id) acc

File: bolychev.li/final.hs (hunk 3 above)
Comment (Transcendentall) [reply bolychev.li/final.hs:26]: functionSublistBeginningElement :: Int -> [Int] -> [Int]
functionSublistBeginningElement a list = (\(x,y) -> if ((a <= x)) then [x] else []) =<< zip list [0..]

**Review by DieNice — APPROVED** [review]
Ok

**Pull request merged**